clang/runtime/dpct-rt/include/dpct/math.hpp

```diff
@@ -712,6 +712,25 @@ template <typename T> sycl::vec<int32_t, 2> extractAndExtend2(T a) {
   return ret;
 }
 
+template <typename T> sycl::vec<int32_t, 4> extractAndExtend4(T a) {
+  sycl::vec<int32_t, 4> ret;
+  sycl::vec<T, 1> va{a};
+  if constexpr (std::is_signed_v<T>) {
+    auto v = va.template as<sycl::vec<int8_t, 4>>();
+    ret[0] = zero_or_signed_extent(v[0], 17);
```

Comment:
comment1: v[0] is int8, so Line 720, we only 9 bit is enough.
comment2:  zero_or_signed_extent() return int64, should here using a zero_or_signed_extent() which return a 32int.

Reply:
fixed